tracer/test/snapshots/NetActivitySdkTests.verified.txt

```diff
@@ -0,0 +1,456 @@
+    SpanId: Id_2,
+    Name: Samples.NetActivitySdk.server,
+    Resource: Ctor1,
+    Service: Samples.NetActivitySdk,
+    Type: web,
+    Tags: {
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: Samples.NetActivitySdk,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      runtime-id: Guid_2,
+      span.kind: server,
+      _dd.p.dm: -0
+    },
+    Metrics: {
+      process_id: 0,
+      _dd.agent_psr: 1.0,
+      _dd.top_level: 1.0,
+      _dd.tracer_kr: 1.0,
+      _sampling_priority_v1: 1.0
+    }
+  },
+  {
+    TraceId: Id_1,
+    SpanId: Id_3,
+    Name: Samples.NetActivitySdk.server,
+    Resource: Ctor2,
+    Service: Samples.NetActivitySdk,
+    Type: web,
+    ParentId: Id_2,
+    Tags: {
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: Samples.NetActivitySdk,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      span.kind: server
+    }
+  },
+  {
+    TraceId: Id_1,
+    SpanId: Id_4,
+    Name: Samples.NetActivitySdk.server,
+    Resource: Ctor3,
+    Service: Samples.NetActivitySdk,
+    Type: web,
+    ParentId: Id_3,
+    Tags: {
+      attribute-string: str,
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: Samples.NetActivitySdk,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      span.kind: server
+    },
+    Metrics: {
+      attribute-int: 1.0
+    }
+  },
+  {
+    TraceId: Id_1,
+    SpanId: Id_5,
+    Name: Samples.NetActivitySdk.server,
+    Resource: Ctor4,
+    Service: Samples.NetActivitySdk,
+    Type: web,
+    ParentId: Id_4,
+    Tags: {
+      attribute-string: str,
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: Samples.NetActivitySdk,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      span.kind: server
+    },
+    Metrics: {
+      attribute-int: 1.0
+    }
+  },
+  {
+    TraceId: Id_6,
+    SpanId: Id_7,
+    Name: Samples.NetActivitySdk.internal,
+    Resource: ErrorSpan,
+    Service: Samples.NetActivitySdk,
+    Type: custom,
+    Error: 1,
+    Tags: {
+      env: integration_tests,
+      error.msg: SetStatus-Error,
+      language: dotnet,
+      otel.library.name: Samples.NetActivitySdk,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_3,
+      runtime-id: Guid_2,
+      span.kind: internal,
+      _dd.p.dm: -0
+    },
+    Metrics: {
+      process_id: 0,
+      _dd.agent_psr: 1.0,
+      _dd.top_level: 1.0,
+      _dd.tracer_kr: 1.0,
+      _sampling_priority_v1: 1.0
+    }
+  },
+  {
+    TraceId: Id_6,
+    SpanId: Id_8,
+    Name: Samples.NetActivitySdk.internal,
+    Resource: OkSpan,
+    Service: Samples.NetActivitySdk,
+    Type: custom,
+    ParentId: Id_7,
+    Tags: {
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: Samples.NetActivitySdk,
+      otel.status_code: STATUS_CODE_UNSET,
```

Comment:
Should this have a `200` status code (based on the Resource name)?

Reply:
The `ActivityStatusCode` is more or less equivalent to our `span.Error` bool but instead it has three states: UNSET, OK, and ERROR. So it doesn't really map to HTTP status codes, even though it kind of looks that way

Reply:
But actually, this should be setting otel.status_code to STATUS_CODE_OK, so I think we might be missing some logic in our ActivityHandler's

Reply:
Thanks for catching this, I'll have a fix for this in this PR too